opensearch/user_data.yaml.tpl

```diff
@@ -39,6 +39,56 @@ write_files:
     permissions: "0400"
     content: |
       ${indent(6, tls.admin_key)}
+%{ if try(snapshot_repository.ca_cert, "") != "" ~}
+  - path: /etc/opensearch/snapshot-repository/ca.crt
+    owner: root:root
+    permissions: "0400"
+    content: |
+      ${indent(6, snapshot_repository.ca_cert)}
+  - path: /usr/local/bin/import_snapshot_repository_ca
+    owner: root:root
+    permissions: "0555"
+    content: |
+      #!/bin/bash
+      set -euo pipefail
+
+      CA_FILE=/etc/opensearch/snapshot-repository/ca.crt
+      CACERTS="/opt/opensearch/jdk/lib/security/cacerts"
+      OS_KEYSTORE="/etc/opensearch/configuration/opensearch.keystore"
+
+      if [ ! -s "$CA_FILE" ]; then
+        exit 0
+      fi
+
+      TMP_PREFIX="/tmp/snapshot-repo-ca"
+      rm -f $${TMP_PREFIX}-*.pem || true
+      awk -v prefix="$TMP_PREFIX" 'BEGIN{i=0} /BEGIN CERTIFICATE/{i++} {print > (prefix "-" i ".pem")}' "$CA_FILE"
+
+      shopt -s nullglob
+      imported=0
+      for cert in $${TMP_PREFIX}-*.pem; do
+        if openssl x509 -in "$cert" -noout -text | grep -q "CA:TRUE"; then
+          alias="snapshot-repository-$(basename "$cert" .pem)"
+          if ! /opt/opensearch/jdk/bin/keytool -list -alias "$alias" -keystore "$CACERTS" -storepass changeit >/dev/null 2>&1; then
+            /opt/opensearch/jdk/bin/keytool -importcert -noprompt \
+              -alias "$alias" \
+              -file "$cert" \
+              -keystore "$CACERTS" \
+              -storepass changeit
+            imported=1
+          fi
+        fi
+      done
+      shopt -u nullglob
+      rm -f $${TMP_PREFIX}-*.pem || true
+
+      chown opensearch:opensearch "$OS_KEYSTORE"
+      chmod 600 "$OS_KEYSTORE"
+
+      if [ "$imported" -eq 1 ]; then
+        systemctl restart opensearch
```

Comment:
You are running this script before opensearch is started. You can simply remove this and make it a sightly easier read. Opensearch will not be started by the time this code is reached.